src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,121 @@
+package com.pusher.client.channel.impl;
+
+import com.google.gson.Gson;
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.channel.ChannelState;
+import com.pusher.client.channel.PrivateEncryptedChannel;
+import com.pusher.client.channel.PrivateEncryptedChannelEventListener;
+import com.pusher.client.channel.SubscriptionEventListener;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.crypto.nacl.SecretBoxOpener;
+import com.pusher.client.util.Factory;
+import com.pusher.client.util.internal.Base64;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PrivateEncryptedChannelImpl extends ChannelImpl implements PrivateEncryptedChannel {
+
+    private static final Gson GSON = new Gson();
```

Comment:
probably something more for the future: that's a heavy object and it could be reused across all currently 8 places in the code base that create it. Indeed depending on the number of channels created it will be many instances duplicated in the memory... also ALL_CAPS-wise it's not a const as that object is mutable

Reply:
reusing could also allow to share common configuration if needed in the future

Reply:
Hmm. So, I made the ChannelImpl GSON protected, and re-used that one. But it's a bit of a lie to say it's re-using, because each channel will create it's own GSON object. Have added this to our list of improvements for the future.